rom-elks.c

```diff
@@ -640,4 +640,11 @@ void rom_init (void)
 	mem_write_byte (0xFFFF0, 0xEA,   1);  // JMPF
 	mem_write_word (0xFFFF1, 0x0000, 1);
 	mem_write_word (0xFFFF3, 0xF000, 1);
+
+	// BIOS Data Area (BDA) setup for EGA/MDA adaptors
+
+	mem_write_byte (BDA_BASE+0x49, 0, 3); 			// video mode (7=MDA)
+	mem_write_byte (BDA_BASE+0x4a, VID_COLS, 1);	// console width
+	mem_write_word (BDA_BASE+0x4c, VID_PAGE_SIZE, 1); // page size
+	mem_write_word (BDA_BASE+0x63, CRTC_IOBASE, 1);	// 6845 CRTC base register
```

Comment:
You could simplify by directly accessing to `mem_stat`.

Reply:
All other code in this file uses `mem_write_xxx`. I didn't like access `mem_stat` directly unless we have to.

Reply:
Moved BDA init here. I'm leaving setting of byte/word values to use mem_write_xxx wrappers because of potential portability problems on RISC processors writing words at byte boundaries (reference word at BDA_BASE+0x63). Then, when EMU86 ported to RISC (if ever), this will be fixed in mem_write_word routine.

Reply:
The functions `mem_write_*` and `mem-read_*` are there to control access to the emulated memory, for device mapping & monitoring purpose, but only from the the sandboxed real code.

For example, we don't want a data breakpoint hit when the emulation code (device or ROM) read or write the emulated memory. I made a design mistake by using these functions  in `rom-*` with a tricky `init` flag to disable access control. They have to be replaced here by direct accesses to `mem_stat`, and the uggly flag `init` is to be removed.

So please, at least for this PR, write the new BDA initialization without call to these functions.